lib/HomeyWidget.d.ts

```diff
@@ -0,0 +1,65 @@
+export = HomeyWidget;
+/**
+ * HomeyWidget is the interface for the Homey instance
+ * used to communicate with Homey from an app widget.
+ * It is the type of the argument in the `onHomeyReady` function.
+ */
+interface HomeyWidget {
+    /**
+     * Translate a string programmatically.
+     *
+     * @example
+     * Homey.__('errors.device_unavailable');
+     * @example
+     * Homey.__({ en: 'My String', nl: 'Mijn tekst' });
+     *
+     * @param {object|string} key translation string or Object
+     * @param {object} [tags] values to interpolate into the translation
+     * @returns {string}
+     */
+    __(key: object | string, tags?: object): string;
+    /**
+     * Make a call to the widget's web API.
+     * @param {'DELETE'|'GET'|'POST'|'PUT'} method
+     * @param {string} uri
+     * @param {object} [body]
+     */
+    api(
+        method: 'DELETE' | 'GET' | 'POST' | 'PUT',
+        uri: string,
+        body?: object,
+    ): Promise<unknown>;
+    /**
+     * Get an object with all settings.
+     * @returns {Object.<string, unknown>}
+     */
+    getSettings(): Record<string, unknown>;
+    /**
+     * Get the unique id for the instance of the widget.
+     * @returns {string}
+     */
+    getWidgetInstanceId(): string;
+    /**
+     * Register an event listener for the app's realtime events.
+     * @param {string} event
+     * @param {Function} callback
+     */
+    on(event: string, callback: Function): void;
+    /**
+     * Open an in app browser view.
+     * @param {string} url
+     */
+    popup(url: string): Promise<void>;
+    /**
+     * The settings view will be hidden until this method has been called.
+     * Use the extra time to make required API calls to prevent flickering on screen.
+     * @param {Object} [args] An object with a height property.
+     * @param {number|string} [args.height] The height of the widget.
+     */
+    ready(args?: { height: number | string }): Promise<void>;
```

Comment:
Documentation says `void`, but according to my test it is a `Promise` (like for the other interfaces).

Reply:
It doesnt return a promise this is the code. So whats going on here :)

`
  ready({ height } = { height: null }) {
    if (this._ready !== true) {
      this._ready = true;
      this._crossFrame.send('ready', { height });
    }
  }
`

Reply:
@jeroenwienk,

You raised your case, I changed it.

As per my test, it seems that there was less flickering with `await` than without.

Is it the same for the `ready` of app settings and the other `ready`?